components/root/root.jsx

```diff
@@ -219,18 +219,17 @@ export default class Root extends React.Component {
         }
     }
 
-    redirectIfNecessary = (props) => {
-        if (props.location.pathname === '/') {
-            if (this.props.noAccounts) {
-                this.props.history.push('/signup_user_complete');
+    static getDerivedStateFromProps(props, state) {
+        if (state.prevProps && props.location.pathname === '/') {
+            if (state.prevProps.noAccounts) {
+                state.prevProps.history.push('/signup_user_complete');
```

Comment:
This should also be done in `componentDidUpdate`. That'll also save us from having to store the previous props